### PR TITLE
Added support for Talos path rewrite for ECS migration

### PR DESCRIPTION
**Pull request by ChrisBratti-VA**

### CODE REVIEW

File: facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/PathRewriteConfigIT.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+@Slf4j
+public class PathRewriteConfigIT {
+  private static RequestSpecification requestSpecification() {
+    SystemDefinitions.Service svc = systemDefinition().facilities();
+    return RestAssured.given().baseUri(svc.url()).port(svc.port()).relaxedHTTPSValidation();
+  }
+
+  @Test
+  void pathIsRewritten() {
```
Comment (bschofield-va) [inline facilities-tests/src/test/java/gov/va/api/lighthouse/facilities/tests/PathRewriteConfigIT.java:25]: you may consider `@ParameterizedTest` with `@ValueSource(strings={ "/facilities/actuator/health", "/va_facilities/actuator/health", "/services/va_facilities/actuator/health"})`

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Bean
+  FilterRegistrationBean<PathRewriteFilter> pathRewriteFilter() {
+    var registration = new FilterRegistrationBean<PathRewriteFilter>();
+    PathRewriteFilter filter =
+        PathRewriteFilter.builder().removeLeadingPath(leadingPaths()).build();
+    registration.setFilter(filter);
+    registration.setOrder(Ordered.LOWEST_PRECEDENCE);
+    registration.addUrlPatterns(filter.removeLeadingPathsAsUrlPatterns());
+    log.info("PathRewriteFilter is enabled.");
```
Comment (bschofield-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java:30]: I'm not saying to make a change ... this is fine as is. In the future, consider enhancing the logging a little bit to help future you out.

Consider these alternatives:
```
2022-01-05 07:57:00 thread-1 g.v.a.l.f.PathRewriteConfig [INFO] PathRewriteFilter is enabled
```

```
2022-01-05 07:57:00 thread-1 g.v.a.l.f.PathRewriteConfig [INFO] Enabling path rewriting for: [ /va_facilities/, /services/va_facilities/, /facilities/ ]
```

The first (current) log statement is
- partially useful (_enabled_), 
- partially redundant (the logging class is automatically printed so we already know it's path rewrite)
- incomplete (it doesn't tell me what paths are being re-written

The second example solves ^ these gaps.
```
log.info("Enabling path rewriting for: {}", leadingPaths());
```

**Review by bschofield-va — DISMISSED** [review]

File: facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Configuration for doing path re-writes. Removes the leading path(load balancer) from the request.
+ */
+@Slf4j
+@Configuration
+public class PathRewriteConfig {
+
+  static List<String> leadingPaths() {
+    return List.of("/va_facilities/", "/services/va_facilities/", "/facilities/");
```
Comment (as-va) [inline facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java:19]: Does this need anything for endpoints on the `/internal` path?

Comment (ChrisBratti-VA) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java:19]: I think we are good here, the `/internal` path is handled by the `@RequestMapping` tag on the internal controller level

Comment (as-va) [reply facilities/src/main/java/gov/va/api/lighthouse/facilities/PathRewriteConfig.java:19]: Nevermind. This rewriting is only for the path portion above the controller level and the `/internal` path is within the application controllers so should be fine.

**Review by as-va — APPROVED** [review]

**Pull request merged**